src/controllers/authController.js

```diff
@@ -13,26 +14,47 @@ const register = async (req, res) => {
     throw new CustomError.BadRequestError('Please provide all required fields');
   }
 
-  const emailAlreadyExists = await User.findOne({ email }); // Check if a user with the email already exists
-  if (emailAlreadyExists) {
-    throw new CustomError.BadRequestError('Email already exists'); // If user exists, throw an error
-  }
+  const session = await mongoose.startSession();
 
-  const isFirstAccount = (await User.countDocuments({})) === 0; // Check if it's the first account
-  const role = isFirstAccount ? 'admin' : 'user'; // Assign a role based on first account or not
+  try {
+    session.startTransaction();
+    const emailAlreadyExists = await User.findOne({ email }); // Check if a user with the email already exists
 
-  const user = await User.create({
-    name,
-    username,
-    email,
-    password,
-    role,
-  }); // Create a new user in the database
-  const tokenUser = createTokenUser(user); // Create a token based on user data
-  attachCookiesToResponse({ res, user: tokenUser }); // Attach the token to cookies and send in the response
+    if (emailAlreadyExists) {
+      throw new CustomError.BadRequestError('Email already exists'); // If user exists, throw an error
+    }
 
-  res.status(StatusCodes.CREATED).json({ user: tokenUser }); // Send a successful response with user data
-  /* 
+    const isFirstAccount = (await User.countDocuments({})) === 0; // Check if it's the first account
+    const role = isFirstAccount ? 'admin' : 'user'; // Assign a role based on first account or not
+
+    const [user] = await User.create(
+      [
+        {
+          name,
+          username,
+          email,
+          password,
+          role,
+        },
+      ],
+      { session }
+    ); // Create a new user in the database
+
+    // Send the welcome email
+    await emailSender.sendWelcomeEmail(user.email, user);
+
+    const tokenUser = createTokenUser(user); // Create a token based on user data
+    attachCookiesToResponse({ res, user: tokenUser }); // Attach the token to cookies and send in the response
+    await session.commitTransaction();
+    res.status(StatusCodes.CREATED).json({ user: tokenUser }); // Send a successful response with user data
+  } catch (error) {
+    await session.abortTransaction();
+    throw error;
+  } finally {
+    session.endSession();
+  }
+};
+/* 
```

Comment:
The swagger code is now _outside_ of the function; so our swagger-autogen tool will no longer know which controller/endpoint it belongs to. Please move it up inside the function; at the end of the function (so between the brackets on line 55 and 56)

Reply:
Swagger code has been moved inside the function register commit f0cf82f1f4446eaffe3143c03fc372f28d9b8648